pandas/tests/indexing/test_loc.py

```diff
@@ -1130,3 +1130,17 @@ def test_loc_with_period_index_indexer():
     tm.assert_frame_equal(df, df.loc[list(idx)])
     tm.assert_frame_equal(df.iloc[0:5], df.loc[idx[0:5]])
     tm.assert_frame_equal(df, df.loc[list(idx)])
+
+
+def test_loc_replace_subset_with_subset():
+    # GH#36424 Should raise a SettingWithCopyError
+    df1 = pd.DataFrame(
+        data=np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
+        columns=["A", "B", "C"],
+        index=[0, 0, 1],
+    )
+    df2 = df1.copy()
+    df2[:] = np.nan
+    # It fail if a SettingWithCopyError is not raised
+    with pytest.raises(pandas.core.common.SettingWithCopyError):
```

Comment:
Can you import this as `import pandas.core.common as com` up top and then use `com.SettingWithCopyError` here (that's the pattern used in other tests)? Also can you match against the error message (see `/tests/frame/indexing/test_indexing.py` for similar examples)?

Reply:
Done

Reply:
Thanks, can you also test the error message (similar tests should show how to do this)?

Reply:
for sure, I'm just wondering if i should match the full message or only the first part ("A value is trying to be set on a copy of a slice from a DataFrame")

Reply:
Just matching that part is fine